kernel/Kconfig

```diff
@@ -150,7 +150,7 @@ config SCHED_CPU_MASK
 config MAIN_STACK_SIZE
 	int "Size of stack for initialization and main thread"
 	default 2048 if COVERAGE_GCOV
-	default 512 if ZTEST && !(RISCV || X86)
+	default 768 if ZTEST && !(RISCV || X86)
```

Comment:
Pls, make this 1024;

768 is not a power-of-2, and Cortex-M builds on ARMv7-M with Userspace will move to 1024 anyway... So putting 1024 will be less confusing.

Alternatively (since your primary target is ARM Cortex-M) you can add a new default line before line 153:

default 1024 if TEST_ARM_CORTEX_M

Reply:
So you won't be changing the default setting for other ARCHes like Xtensa, Sparc, AARC64 and ARC